cmd.go

```diff
@@ -20,124 +20,124 @@ import (
+	fmt.Println(gotext.Get("    --rebuildtree         Always build all AUR packages even if installed"))
+	fmt.Println(gotext.Get("    --redownload          Always download pkgbuilds of targets"))
+	fmt.Println(gotext.Get("    --noredownload        Skip pkgbuild download if in cache and up to date"))
+	fmt.Println(gotext.Get("    --redownloadall       Always download pkgbuilds of all AUR packages"))
+	fmt.Println(gotext.Get("    --provides            Look for matching providers when searching for packages"))
+	fmt.Println(gotext.Get("    --noprovides          Just look for packages by pkgname"))
+	fmt.Println(gotext.Get("    --pgpfetch            Prompt to import PGP keys from PKGBUILDs"))
+	fmt.Println(gotext.Get("    --nopgpfetch          Don't prompt to import PGP keys"))
+	fmt.Println(gotext.Get("    --useask              Automatically resolve conflicts using pacman's ask flag"))
+	fmt.Println(gotext.Get("    --nouseask            Confirm conflicts manually during the install"))
+	fmt.Println(gotext.Get("    --combinedupgrade     Refresh then perform the repo and AUR upgrade together"))
+	fmt.Println(gotext.Get("    --nocombinedupgrade   Perform the repo upgrade and AUR upgrade separately"))
+	fmt.Println(gotext.Get("    --batchinstall        Build multiple AUR packages then install them together"))
+	fmt.Println(gotext.Get("    --nobatchinstall      Build and install each AUR package one by one"))
+	fmt.Println()
+	fmt.Println(gotext.Get("    --sudo                <file>  sudo command to use"))
+	fmt.Println(gotext.Get("    --sudoflags           <flags> Pass arguments to sudo"))
+	fmt.Println(gotext.Get("    --sudoloop            Loop sudo calls in the background to avoid timeout"))
+	fmt.Println(gotext.Get("    --nosudoloop          Do not loop sudo calls in the background"))
+	fmt.Println()
+	fmt.Println(gotext.Get("    --timeupdate          Check packages' AUR page for changes during sysupgrade"))
+	fmt.Println(gotext.Get("    --notimeupdate        Do not check packages' AUR page for changes"))
+	fmt.Println()
+	fmt.Println(gotext.Get("show specific options:"))
+	fmt.Println(gotext.Get("    -c --complete         Used for completions"))
+	fmt.Println(gotext.Get("    -d --defaultconfig    Print default yay configuration"))
+	fmt.Println(gotext.Get("    -g --currentconfig    Print current yay configuration"))
+	fmt.Println(gotext.Get("    -s --stats            Display system package statistics"))
+	fmt.Println(gotext.Get("    -w --news             Print arch news"))
+	fmt.Println()
+	fmt.Println(gotext.Get("yay specific options:"))
+	fmt.Println(gotext.Get("    -c --clean            Remove unneeded dependencies"))
+	fmt.Println(gotext.Get("       --gendb            Generates development package DB used for updating"))
+	fmt.Println()
+	fmt.Println(gotext.Get("getpkgbuild specific options:"))
+	fmt.Println(gotext.Get("    -f --force            Force download for existing ABS packages"))
```

Comment:
Genuine question: Is this the way other projects translate their usage sections?

Why can't it remain one string, which, as a whole, should be translated?

Reply:
Hello

I am using "yay" conveniently. Thanks.
I will explain the reason why it is divided into lines.

Creating a POT file in page-by-page display is incompatible and makes the "xgettext" tool unusable. For example, "syntax error" and "keyword unknown" occur frequently in "msginit" and "msgfmt". I think the advantage of using POT, PO, and MO files when translating is that you can use the "xgettext" tool.



## Operation example:

- I created a POT file using "xgotext". It is the operation after that.

```
$ cd ~/go/src/yay-help/out/

$ cat yay-hlp3.pot
msgid ""
msgstr ""
"Plural-Forms: nplurals=2; plural=(n != 1);\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Language: \n"
"X-Generator: xgotext\n"

#: cmd.go:23
msgid `Usage:
    yay
    yay <operation> [...]
    yay <package(s)>

: (Omitted)

getpkgbuild specific options:
    -f --force            Force download for existing ABS packages`
msgstr ""
```



### Try using "msginit":

```
$ LANG=C msginit --locale=en --input=yay-hlp3.pot
out/yay-hlp3.pot:11:8: syntax error
out/yay-hlp3.pot:11: keyword "Usage" unknown
out/yay-hlp3.pot:12: keyword "yay" unknown
out/yay-hlp3.pot:13: keyword "yay" unknown

: (Omitted)

msginit: too many errors, aborting
```



### Try using "msgfmt":

```
$ cp yay-hlp3.pot yay-hlp3.po

$ LANG=C msgfmt -o yay-hlp3.mo yay-hlp3.po
yay-hlp3.po:11:8: syntax error
yay-hlp3.po:11: keyword "Usage" unknown
yay-hlp3.po:12: keyword "yay" unknown

: (Omitted)

msgfmt: too many errors, aborting
```

Reply:
Hey @Jguer , thank you for your hard work.

I think it's okay if the placement information is embedded in the translation. If you change it too much, the source program will be difficult to see. Translation is only difficult at first. It is not updated every time. Instead, it is more efficient to focus on other things.

I'm editing a PO file directly in a text editor, so it's easier to translate if I can see the whole thing rather than just a few strings.

If you paste the "help display part" added to "en.po" into a PO file in another language, the "msgid" will be copied to the "msgstr", making it easier to understand the translation position. It might have been better to do so.